src/parser.rs

```diff
@@ -87,14 +91,17 @@ fn ast_of_parse_tree(pairs: Pairs<Rule>) -> Result<Items> {
                 }
                 Rule::table => {
                     let parsed = ast_of_parse_tree(pair.into_inner())?;
-                    if let [name, Item::Pipeline(pipeline)] = &parsed[..] {
-                        Item::Table(Table {
-                            name: name.as_ident()?.to_owned(),
-                            pipeline: pipeline.clone(),
-                        })
-                    } else {
-                        unreachable!("Expected Table, got {:?}", parsed)
-                    }
+                    let [name, pipeline]: [Item; 2] = parsed
+                        .try_into()
+                        .map_err(|e| anyhow!("Expected two items; {:?}", e))?;
+                    Item::Table(Table {
+                        name: name
+                            .into_ident()
+                            .map_err(|e| anyhow!("Expected Ident; {:?}", e))?,
+                        pipeline: pipeline
+                            .into_pipeline()
+                            .map_err(|e| anyhow!("Expected Pipeline; {:?}", e))?,
```

Comment:
@kwigley do you know whether the TODO you added around deriving our own Error type would make these terser? There are lots of them now... (I'm less familiar with these "newer" error-handling crates in rust...)

Reply:
I think this is possibly a different issue: https://github.com/bluejekyll/enum-as-inner/issues/84